Cython/Compiler/UFuncs.py

```diff
@@ -138,13 +138,17 @@ def generate_cy_utility_code(self):
 
         ufunc_cname = self.global_scope.next_id(self.node.entry.name + "_ufunc_def")
 
+        outer_is_nogil = not (any(t.is_pyobject for t in arg_types) or
+            any(t.is_pyobject for t in out_types))
```

Comment:
`outer_is_nogil` suggests to me that there is an outer function that was declared `nogil` (by the user). However, the intention seems more like `as_nogil_function` or `can_run_without_gil` or `needs_gil`, or `reacquire_gil` or something like that.